include/tvm/tir/function.h

```diff
@@ -240,7 +240,7 @@ namespace attr {
  *
  * Call(f,
  *      [arg1, arg2, ..., arg_n,
- *       work_size_1, work_size_2, ... work_size_m])
+ *       work_size_1, work_size_2, ... work_size_m, dyn_shmem_size])
```

Comment:
Let us introduce a special meta-data to indicate that dynamic shared memory is used. This is to make sure the calling convention is backward compatible when dyn shared memory is not provided.

```
constexpr const char* kDeviceUseDynSharedMemory = "tir.device_use_dyn_shared_memory";
```

Reply:
done